### ISSUE TITLE
[MNG-2817] Add identity specification in maven-model and maven-settings

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#4912]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** opened **[MNG-2817](https://issues.apache.org/jira/browse/MNG-2817?redirect=false)** and commented

Some generated objects are used in lists. Thus, it will be very useful to have identity specification for them: equals(..) and hashcode() (see MODELLO-43)
For instance, see org.apache.maven.model.Resource used in model/build/resources

---

**Issue Links:**
- [MNG-3265](https://issues.apache.org/jira/browse/MNG-3265) maven-model Extension.equals causes NPE when any field is uninitialized

- [MNG-3266](https://issues.apache.org/jira/browse/MNG-3266) maven-model RepositoryBase overrides equals() but not hashCode()

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

Modello already generates these methods, just add the identifier parameter:

```
<field>
  <name>name</name>
  <version>3.0.0+</version>
  <type>String</type>
  <identifier>true</identifier>
</field>
```

**Comment by jira-importer** [comment]
**[Siveton Vincent](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=siveton)** commented

Waiting for identities from the super class.

**Comment by jira-importer** [comment]
**[jroberts](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jroberts)** commented

Is there any way this could get patched back to the 2.0.x branch? These are defect fixes rather than new functionality and the project I'm working on would rather not wait until the 2.1 release.

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

Consolidating to 2.1.0-M1 so we can then rename to 2.1.0. We can weed out any issues we want to push to a later release from this set once we've done the consolidation.
